Bottom: 966ce34c2dc0841f7a3beb929f93b56f46904b09
Top:    9cea3990b723d805583c621d3b0ae1a1f7d4263c
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-17 12:35:20 +0800

autofs-5.0.7 - use ulimit max open files if greater than internal maximum

When setting the maximum number of allowed file handles the current setting
should be checked before setting it. If the ulimit command has been used to
increase the maximum to larger than what automount would ask for then honour
it.


---

diff --git a/daemon/automount.c b/daemon/automount.c
index 019637f..1d0b64e 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -2106,8 +2106,11 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
-	rlim.rlim_cur = MAX_OPEN_FILES;
-	rlim.rlim_max = MAX_OPEN_FILES;
+	res = getrlimit(RLIMIT_NOFILE, &rlim);
+	if (res == -1 || rlim.rlim_max <= MAX_OPEN_FILES)  {
+		rlim.rlim_cur = MAX_OPEN_FILES;
+		rlim.rlim_max = MAX_OPEN_FILES;
+	}
 	res = setrlimit(RLIMIT_NOFILE, &rlim);
 	if (res)
 		printf("%s: can't increase open file limit - continuing",
